ext/bigdecimal/bigdecimal.c

```diff
@@ -2681,9 +2681,8 @@ VpNewVarArg(int argc, VALUE *argv)
  *           If it is a String, spaces are ignored and unrecognized characters
  *           terminate the value.
  *
- * digits:: The number of significant digits, as an Integer. If omitted or 0,
- *          the number of significant digits is determined from the initial
- *          value.
+ * digits:: The number of significant digits, as an Integer. This is required if
+ *          +initial+ is a Float or Rational, and ignored otherwise.
```

Comment:
`0` for `digits` has the special meaning that the number of significant digits is automatically determined from the given initial value.  For a Rational initial value, for example, the resulting BigDecimal is determined on the value of `BigDecimal.limit` and the behavior of `BigDecimal#div`.

Reply:
OK.  I wonder if there is a better way to word it.  I want the documentation to explain that it does not have an effect in cases where you would expect it would:

```ruby
BigDecimal(1.1234, 2)
# => 0.11e1
BigDecimal(Rational(11234,10000), 2)
# => 0.11e1
BigDecimal('1.1234', 2)
# => 0.11234e1

BigDecimal(11234.0, 2)
# => 0.11e5
BigDecimal(11234, 2)
# => 0.11234e5
```